### PR TITLE
build: bump flutter and dependencies versions

### PR DESCRIPTION
**Pull request by adam-sroka**
Bump flutter version to the new major [3.0.0 release](https://medium.com/flutter/whats-new-in-flutter-3-8c74a5bc32d0). Update all dependencies to newest versions, including breaking major version bumps. Fix resulting conflicts in [Dropdown Search](https://pub.dev/packages/dropdown_search), albeit still without fixing #41.

I opened an [Issue](https://github.com/salim-lachdhaf/searchable_dropdown/issues/407) in Dropdown search repo, but this is something for another PR, and is still tracked in #41.

tl;dr This PR tries to get up to speed with all sdks and deps, so that we can implement stuff efficiently and be up to date on everything 🙌

### CODE REVIEW

**Comment by adam-sroka** [comment]
@sshazril lmk if this breaks anything else, I didn't spot anything but you never know. Also, we can merge this before or after #47, based on who's more willing to fix some merge conflicts 😛.

**Comment by adam-sroka** [comment]
Aight, should be ok now.
